math/pollard_test.go

```diff
@@ -0,0 +1,79 @@
+// pollard_test.go
+// description: Test for Pollard's rho algorithm
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see pollard.go
+
+package math
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+)
+
+func TestDefaultPolynomial(t *testing.T) {
+	tests := []struct {
+		name string
+		x    *big.Int
+		n    *big.Int
+		want *big.Int
+	}{
+		{"Polynomial for n = 1772; x = 535", big.NewInt(535), big.NewInt(1772), big.NewInt(934)},
+		{"Polynomial for n = 666; x = 53135", big.NewInt(53135), big.NewInt(666), big.NewInt(380)},
+		{"Polynomial for n = 666; x = 13", big.NewInt(13), big.NewInt(666), big.NewInt(170)},
+		{"Polynomial for n = 1917; x = 2510", big.NewInt(2510), big.NewInt(1917), big.NewInt(839)},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := DefaultPolynomial(test.x, test.n); !reflect.DeepEqual(got, test.want) {
+				t.Errorf("DefaultPolynomial() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestRho(t *testing.T) {
+	tests := []struct {
+		name    string
+		n       *big.Int
+		g       func(*big.Int, *big.Int) *big.Int
+		want    *big.Int
+		wantErr bool
+	}{
+		{"Rho for n:  11235 ", big.NewInt(11235), DefaultPolynomial, big.NewInt(21), false},
```

Comment:
```suggestion
		{"Factor of n:  11235 ", big.NewInt(11235), DefaultPolynomial, big.NewInt(21), false},
```
I think this naming convention will be more intutive.

Reply:
Done